cclib/method/ddec.py

```diff
@@ -88,17 +96,34 @@ def _read_proatom(self, directory, atom_num, charge):
         #       atoms.h5
         # File format:
         #   Starting from line 13, each line contains the charge densities for each shell
-        chargemol_path = os.path.join(
+        charge_floor = int(math.floor(charge))
```

Comment:
Can you explain where this comes from?

Reply:
From step 1 of the algorithm, non-integer reference charges will be assigned to each atom. So in step 2 of the algorithm, the proatom densities have to be linearly interpolated between the densities of the ion/atom with floor(charge) and ceiling(charge). Let me know if I missed anything. Thanks.

Reply:
Can you add this as a code comment right above?

Reply:
Yes -- I've just added it as a comment. Thanks!